newIDE/app/src/EventsFunctionsExtensionEditor/OptionsEditorDialog/ExtensionDependenciesEditor.js

```diff
@@ -0,0 +1,168 @@
+import Paper from '@material-ui/core/Paper';
+import FlatButton from '../../UI/FlatButton';
+import IconButton from '../../UI/IconButton';
+import Add from '@material-ui/icons/Add';
+import Delete from '@material-ui/icons/Delete';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+import SelectField from '../../UI/SelectField';
+import SelectOption from '../../UI/SelectOption';
+
+import Window from '../../Utils/Window';
+import { mapVector } from '../../Utils/MapFor';
+import newNameGenerator from '../../Utils/NewNameGenerator';
+import useForceUpdate from '../../Utils/UseForceUpdate';
+import BackgroundText from '../../UI/BackgroundText';
+
+const checkNameExists = (
+  name: string,
+  deps: gdVectorDependencyMetadata
+): boolean => {
+  for (let i = 0; i < deps.size(); i++)
+    if (deps.at(i).getName() === name) return true;
+  return false;
+};
+
+type Props = {| eventsFunctionsExtension: gdEventsFunctionsExtension |};
+
+export const ExtensionDependenciesEditor = ({
+  eventsFunctionsExtension,
+}: Props) => {
+  const deps = eventsFunctionsExtension.getAllDependencies();
+  const forceUpdate = useForceUpdate();
+
+  const addDependency = () => {
+    eventsFunctionsExtension
+      .addDependency()
+      .setName(
+        newNameGenerator('New Dependency', newName =>
+          checkNameExists(newName, deps)
+        )
+      )
+      .setExportName('my-dependency')
+      .setVersion('1.0.0')
+      .setDependencyType('cordova');
+    forceUpdate();
+  };
+
+  return (
+    <>
+      <TableContainer
+        component={({ children }) => <Paper elevation={4}>{children}</Paper>}
+      >
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell>
+                <Trans>Name</Trans>
+              </TableCell>
+              <TableCell>
+                <Trans>Export name</Trans>
+              </TableCell>
+              <TableCell>
+                <Trans>Version</Trans>
+              </TableCell>
+              <TableCell>
+                <Trans>Dependency type</Trans>
+              </TableCell>
+              <TableCell>
+                <Trans>Action</Trans>
+              </TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {mapVector<gdDependencyMetadata, TableRow>(
+              eventsFunctionsExtension.getAllDependencies(),
+              (dependency, index) => (
+                <TableRow key={dependency.getName()}>
+                  <TableCell>
+                    <SemiControlledTextField
+                      commitOnBlur
+                      value={dependency.getName()}
+                      onChange={newName => {
+                        if (checkNameExists(newName, deps))
+                          Window.showMessageBox(
+                            `This name is already in use! Please use a unique name.`,
+                            'error'
+                          );
```

Comment:
Prefer `showWarningBox` for warnings that interrupt the workflow (but not an error, because nothing dangerous happened, it's a mere explanation that you can't go further).
Also pass `delayToNextTick` to true to avoid some weird freeze/infinite loop of the app on some linux distro because of a blur + a message box displayed at the same time.